### PR TITLE
File storage without additional tasks [Work in progress]

### PR DESCRIPTION
**Pull request by Werocky1502**
Partially finished third task. Review required to make corrections.

All commands work, the description can be found in the README.md in the project root.

In the process of improving the part with parsing incoming text commands (will be rewritten later).

### CODE REVIEW

File: Lab-file-storage/Storage/StorageManager.cs [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+                return unit.FullMetaUnitInfo;
+            }
+            else
+            {
+                throw new FileLoadException("No such file in storage!");
+            }
+        }
+
+        public string GetUserInfo()
```
[the remark this answers is not in the document]
Comment (Werocky1502) [reply Lab-file-storage/Storage/StorageManager.cs]: Method itself and user field were removed from StorageManager. Now information about user shown using public property in StorageManager and UserInfoCommand, which combine info from storage and from user.

Nevertheless, I'm add Login string to file upload method cause I need it during MetaFileUnit creation.

Actually I'm not shure that problem was solved.

File: Lab-file-storage/InternalCommandShell.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        private string GetUserInput()
+        {
+            _userInteraction.PrintSystemMessage("Enter command: ");
+
+            return _userInteraction.GetUserInput();
+        }
+
+        private SplitedArgs ParseUserInput(string inputString)
+        {
+            List<string> splitedInput = inputString.Split('"').ToList();
```
Comment (HenadziBaranau) [inline Lab-file-storage/InternalCommandShell.cs]: Why do we need .ToList here?

File: Lab-file-storage/Utilities/DateAndTime/UTCDateTimeProvider.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace Lab_file_storage.Utilities.DateAndTime
+{
+    public class UtcDateTimeProvider : IDateTimeProvider
+    {
+        public DateTime CurrentTimeAndDate { get { return DateTime.UtcNow; } }
```
Comment (HenadziBaranau) [inline Lab-file-storage/Utilities/DateAndTime/UTCDateTimeProvider.cs]: Not a mistake, but I would prefer c# 6 property syntax as lambda expression when possible:)

File: Lab-file-storage/Users/User.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        public string Login { get; }
+
+        public DateTime CreationDate { get; }
+
+        public string UserInfo
+        {
+            get
+            {
+                string date = CreationDate.Date.ToString("yyyy-MM-dd");
+                return string.Format($"Login: {Login}\nCreation date: {date}");
```
Comment (HenadziBaranau) [inline Lab-file-storage/Users/User.cs]: string.Format is not needed here

File: Lab-file-storage/Users/AuthenticationService.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        }
+
+        public AuthResult RequestAccess(string login, string password)
+        {
+            if ((login == userConfig.UserName) && (password == userConfig.Password))
+            {
+                _currentUser = new User(userConfig.UserName, userConfig.CreationDate);
+                return new AuthResult(true, _currentUser);
+            }
+            else
```
Comment (HenadziBaranau) [inline Lab-file-storage/Users/AuthenticationService.cs]: redundant else

File: Lab-file-storage/Users/AuthResult.cs [hunk 6]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Lab_file_storage.Users
+{
+    public class AuthResult
```
Comment (HenadziBaranau) [inline Lab-file-storage/Users/AuthResult.cs]: I would recommend to create 2 static functions here CreateSuccessful(user), CreateUnsuccessful() and private ctor to limit user of our class creation of unsuccessful result with user or successful without(null)

File: Lab-file-storage/Storage/StorageManager.cs [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+                    string metaUnitInfo = unit.GeneralMetaUnitInfo;
+                    return $"File \"{filePath}\" has been uploaded.\n{metaUnitInfo}";
+                }
+                catch (FileLoadException)
+                {
+                    File.Delete(newFilePath);
+                    throw;
+                }
+            }
+            else
```
Comment (HenadziBaranau) [inline Lab-file-storage/Storage/StorageManager.cs]: redundant else, and there are multiple below.. just my preferences :)

File: Lab-file-storage/Storage/StorageManager.cs [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+        }
+
+        public string GetUserInfo()
+        {
+            string userInfo = _currentUser.UserInfo;
+            string usedStorage = FileSizeHumanReadable.GetHumanReadableFileSize(_metaFile.UnitStorageSizeInBytes);
+            return $"{userInfo}\nStorage used: {usedStorage}";
+        }
+
+        public bool IsFilesEqual(string fileName, string newFilePath)
```
Comment (HenadziBaranau) [inline Lab-file-storage/Storage/StorageManager.cs]: AreFilesEqual and could be private

File: Lab-file-storage/MetaFile/IMetaFileManager.cs [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+        void AddUnit(MetaFileUnit unit);
+
+        void RemoveUnit(string name);
+
+        MetaFileUnit GetUnitByName(string name);
+
+        MetaFileUnit PopUnitWithName(string name);
+
+        bool IsUnitExist(string name);
```
Comment (HenadziBaranau) [inline Lab-file-storage/MetaFile/IMetaFileManager.cs]: DoesUnitExist? :)

File: Lab-file-storage/InternalCommandShell.cs (hunk 2 above)
Comment (Werocky1502) [reply Lab-file-storage/InternalCommandShell.cs]: You're right, List is not used here. I will remove it.

File: Lab-file-storage/Users/User.cs (hunk 4 above)
Comment (Werocky1502) [reply Lab-file-storage/Users/User.cs]: It is left from the old version of the code. I refactored without thinking. In the future I will pay more attention to this.

File: Lab-file-storage/Users/AuthResult.cs (hunk 6 above)
Comment (Werocky1502) [reply Lab-file-storage/Users/AuthResult.cs]: I got it. Actually it the same approach as in our ExecutionResult in Commands

**Pull request merged**